### PR TITLE
#161243899 Build out REST Endpoints for Sales Endpoint

### PR DESCRIPTION
**Pull request by KMaina**
#### What does this PR do?

Allow a user create sales records and fetch them

#### Description of Task to be completed?
Have the following endpoints working

`POST /api/v1/auth/signup`

`GET /api/v1/users`

`GET /api/v1/users/<int:user_id>`

`POST /api/v1/products`

`GET /api/v1/products`

`GET /api/v1/products/<product_id>`

`POST /api/v1/sales`

`GET /api/v1/sales`

`GET /api/v1/sales/<int:sales_id>`

#### How should this be manually tested?

- clone the repo locally
 
- cd into the Store-Manager-API folder
 
- run git pull https://github.com/KMaina/Store-Manager-API.git ft-sales-resource-endpoints-161243899
 
- in a terminal run the command pytest

- ensure that the following environment variables are set
```
FLASK_ENV=development
FLASK_APP=run.py
SECRET_KEY=<anything-of-choice>
```

- in a terminal run flask run to start the server

- in Postman use the following JSON body to sign up
```
{
   "name" : "Ken",
   "password" : "pass1234",
   "confirm" : "pass1234"
}
```

- in Postman use the following JSON body to log in
```
{
   "name" : "Ken",
   "password" : "pass1234"
}
```

- in Postman use the following JSON body to add a product

```
{
   "name" : "Bread",
   "quantity" : 30,
   "price" : 50,
   "reorder"  : 20 
}
```

- in Postman use the following JSON body to add a sale

```
{
   "product" : "Bread",
   "quantity" : 30,
   "price" : 50
}
```
#### Any background context you want to provide?
This is all functionality for the sales, users and products endpoints
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/161243899

### CODE REVIEW

**Review by Ruiru11 — COMMENTED** [review]
I don't see anything wrong
But i can see there are functions without docstrings 
I think its great if you use them uniformly
(In the user class ass the login method)

**Review by Ruiru11 — COMMENTED** [review]
I don't see anything wrong
But i can see there are functions without docstrings 
I think its great if you use them uniformly
(In the user class ass the login method)

File: app/tests/v1/test_users.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+    def test_fetch_all_users(self):
+        """Tests for fetching all users"""
+        response = self.client().get('/api/v1/users')
+        self.assertEqual(response.status_code, 200)
+
+    def test_fetch_a_specific_user(self):
+        """Tests for fetching a specific user"""
+        response = self.client().post('/api/v1/auth/signup', data=json.dumps(self.user), content_type='application/json')
+        print(response.data)
```
Comment (ghost) [inline app/tests/v1/test_users.py:66]: Why are you printing `response.data` here?

**Comment by ghost** [comment]
@KMaina You don't have to include how to set up the app from scratch in every PR. You can just include how to test that particular feature. Have your set up instructions in your README where they can later be referenced.

**Comment by ghost** [comment]
Format your link to PT like this:
`[#161243899](https://www.pivotaltracker.com/story/show/161243899)`
so it'll be [#161243899](https://www.pivotaltracker.com/story/show/161243899)

File: app/tests/v1/test_users.py (hunk 1 above)
Comment (KMaina) [reply app/tests/v1/test_users.py:66]: That was for debugging, I forgot to take it out

**Pull request merged**